README.md

```diff
@@ -42,8 +42,45 @@ data:
 
 ## Usage
 
+### Supported Backends
+As of today argocd-vault-plugin supports HashiCorp Vault and IBM Secret Manger as backends.
+
+#### HashiCorp Vault
+We support AppRole and Github Auth Method for getting secrets from Vault.
+
+##### AppRole Authentication
+For AppRole Authentication, these are the required environment variables:
```

Comment:
I think we should remove the `AVP_` prefix, rephrase as "required parameters" and then in the next section clearly explain the 3 ways to pass parameters: env vars, config file, k8s secret